File: 89 Gray Code/solution.py
class Solution(object):
    def grayCode(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        upper = 1 << n
        result = []
        for i in range(upper):
            cand = (i >> 1) ^ i
            result.append(cand)
        return result
        

File: 89 Gray Code/solution.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        if (n < 0) {
            return {};
        }
        vector<int> result;
        int upper = 1 << n;
        for (int i = 0; i < upper; i++) {
            result.push_back((i >> 1) ^ i);
        }
        return result;
        
    }
};